### PR TITLE
Override keyword 1472

### PR DESCRIPTION
**Pull request by drmason789**
#1472

I initially added this:
`QMAKE_CXXFLAGS += -Werror=suggest-override`
to the defaults.pri file to make a missing override keyword a compiler error but this causes errors in various other libaries such as Qt, Google Test, so I've commented it out for now. 

There are #pragma commands that can be used to suppress this apparently but I'll look to apply that later.

### CODE REVIEW

File: src/Computrainer.h [hunk 1]
```diff
@@ -145,7 +145,7 @@ class Computrainer : public QThread {
     double getLoad();
 
   private:
-    void run(); // called by start to kick off the CT comtrol thread
+    void run() override; // called by start to kick off the CT comtrol thread
```
Comment (drmason789) [inline src/Computrainer.h:148]: I'm not sure why this is private, it overrides a protected member.

**Comment by drmason789** [comment]
The changes I've made here in this PR so far are those recommended by QtCreator.

**Comment by cagnulein** [comment]
@drmason789 do you need to do other stuff here or I can try for a merge?

**Comment by drmason789** [comment]
@cagnulein You could merge it. If we want to activate the compiler warning or error for missing override keyword though, I believe it needs a certain #pragma directive applied to every #include of group of #includes from 3rd parties (e.g. Qt) where the keyword is not always used. I prefer to merge now and do that in another issue.

**Comment by cagnulein** [comment]
> I initially added this:
> `QMAKE_CXXFLAGS += -Werror=suggest-override`
> to the defaults.pri file to make a missing override keyword a compiler error but this causes errors in various other libaries such as Qt, Google Test, so I've commented it out for now.

do you mean this kind of issues?

```
../../Qt/5.15.2/ios/include/QtCore/qcoreapplication.h:78:5: error: 'metaObject' overrides a member function but is not marked 'override' [-Werror,-Wsuggest-override]
../../Qt/5.15.2/ios/include/QtCore/qcoreapplication.h:78:5: error: 'qt_metacast' overrides a member function but is not marked 'override' [-Werror,-Wsuggest-override]
../../Qt/5.15.2/ios/include/QtCore/qcoreapplication.h:78:5: error: 'qt_metacall' overrides a member function but is not marked 'override' [-Werror,-Wsuggest-override]
../../Qt/5.15.2/ios/include/QtCore/qtimer.h:59:5: error: 'metaObject' overrides a member function but is not marked 'override' [-Werror,-Wsuggest-override]
```

**Comment by drmason789** [comment]
> > I initially added this:
> > `QMAKE_CXXFLAGS += -Werror=suggest-override`
> > to the defaults.pri file to make a missing override keyword a compiler error but this causes errors in various other libaries such as Qt, Google Test, so I've commented it out for now.
> 
> do you mean this kind of issues?
> 
> ```
> ../../Qt/5.15.2/ios/include/QtCore/qcoreapplication.h:78:5: error: 'metaObject' overrides a member function but is not marked 'override' [-Werror,-Wsuggest-override]
> ../../Qt/5.15.2/ios/include/QtCore/qcoreapplication.h:78:5: error: 'qt_metacast' overrides a member function but is not marked 'override' [-Werror,-Wsuggest-override]
> ../../Qt/5.15.2/ios/include/QtCore/qcoreapplication.h:78:5: error: 'qt_metacall' overrides a member function but is not marked 'override' [-Werror,-Wsuggest-override]
> ../../Qt/5.15.2/ios/include/QtCore/qtimer.h:59:5: error: 'metaObject' overrides a member function but is not marked 'override' [-Werror,-Wsuggest-override]
> ```

Yes.

**Pull request merged**